### PR TITLE
[immutable-x] Add Immutable X strategy

### PR DESCRIPTION
**Pull request by sergerad**
Changes proposed in this pull request:
- New strategy under `immutable-x`

### CODE REVIEW

File: src/strategies/immutable-x/index.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    addresses: string[],
+    block: number | string
+): Promise<Record<string, number>> {
+    const multi = new Multicaller(network, provider, abi, { blockTag: block });
+    addresses.forEach((address: string) => multi.call(address, options.address, 'balanceOf', [address]));
+    const result: Record<string, BigNumberish> = await multi.execute();
+    return mapL1Response(result, options);
+}
+
+async function getL2Balances(network: string, options: Options, addresses: string[], block: number | string): Promise<Record<string, number>> {
```
Comment (sergerad) [inline src/strategies/immutable-x/index.ts:67]: @ChaituVR This strategy adds L2 and L1 balances. Did you mean I should leverage another strategy within this one for L1 only?

Comment (ChaituVR) [reply src/strategies/immutable-x/index.ts:52]: Hi @sergerad, No, I mean in this case you can use multichain strategy and pass two `erc20-balance-of` strategies into it 

More details here: https://snapshot.org/#/strategy/multichain
Play with it more and let me know if it multichain strategy doesn't work for you

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/immutable-x/index.ts (hunk 1 above)
Comment (bonustrack) [reply src/strategies/immutable-x/index.ts:52]: The voting strategy "multichain" would only work if the chain is supported. If you want to get voting power from StarkEx for example I would recommend keep this approach.

Comment (sergerad) [reply src/strategies/immutable-x/index.ts:52]: I have used `erc20-balance-of` strategy for L1 in this PR 👍

Comment (sergerad) [reply src/strategies/immutable-x/index.ts:52]: @bonustrack PR is labelled with `changes requested` but my interpretation of your comment was to keep the current impl

Comment (ChaituVR) [reply src/strategies/immutable-x/index.ts:52]: @sergerad Can you revert changes from `yarn.lock` file please, everything else looks good

Comment (sergerad) [reply src/strategies/immutable-x/index.ts:52]: @ChaituVR done, thanks

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**